src/main/java/com/boot/product/dto/ProductDTO.java

```diff
@@ -47,6 +48,8 @@ public class ProductDTO {
 	@Positive(message = "Product stock should be positive number!")
 	private int stock;
 
+	private CharacteristicsDTO characteristics;
```

Comment:
add also notnull validation here

Reply:
have you tried this validation? I had to add also @Valid for the order entries in order for the properties to be validated. Please check!